src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -705,3 +705,272 @@ def get_decoder_prompt_ids(self, task=None, language=None, no_timestamps=True):
+                    "There was an error while processing timestamps, we haven't found a timestamp as last token. Was"
+                    " WhisperTimeStampLogitsProcessor used?"
+                )
+            # Happens when we don't use timestamps
+            resolved_tokens = _find_longest_common_sequence(previous_tokens)
+            resolved_text = self.decode(resolved_tokens)
+            chunk["text"] = resolved_text
+            chunks.append(chunk)
+
+        # Preparing and cleaning up the pipeline output
+        full_text = "".join(chunk["text"] for chunk in chunks)
+        if return_timestamps or return_language:
+            for chunk in chunks:
+                if not return_timestamps:
+                    chunk.pop("timestamp")
+                else:
+                    chunk["timestamp"] = tuple(chunk["timestamp"])
+                if not return_language:
+                    chunk.pop("language")
+            optional = {"chunks": chunks}
+        else:
+            optional = {}
+        return full_text, optional
+
+
+def _find_longest_common_sequence(sequences):
+    # It would be much harder to do O(n) because of fault tolerance.
+    # We actually have a really good property which is that the total sequence
+    # MUST be those subsequences in order.
+    left_sequence = sequences[0]
+    left_length = len(left_sequence)
+    total_sequence = []
+    for right_sequence in sequences[1:]:
+        # index = 0
+        max_ = 0.0
+        max_indices = (left_length, left_length, 0, 0)
+        # Here we're sliding matches
+        # [a, b, c, d]
+        #          [c, d, f]
+        # =        [c] == [d]
+        #
+        # [a, b, c, d]
+        #       [c, d, f]
+        # =     [c, d] == [c, d]
+        #
+        #
+        # [a, b, c, d]
+        #    [c, d, f]
+        #
+        # =  [b, c, d] == [c, d, f]
+        #
+        # [a, b, c, d]
+        # [c, d, f]
+        #
+        # [a, b, c] == [c, d, f]
+        #
+        # [a, b, c, d]
+        # [d, f]
+        #
+        # [a, b] == [d, f]
+        #
+        # [a, b, c, d]
+        # [f]
+        #
+        # [a] == [f]
+        right_length = len(right_sequence)
+        for i in range(1, left_length + right_length):
+            # epsilon to favor long perfect matches
+            eps = i / 10000.0
+
+            # Slightly convoluted because we don't want out of bound indices
+            # This will be necessary for a small conflict resolution optimization
+            # later
+            left_start = max(0, left_length - i)
+            left_stop = min(left_length, left_length + right_length - i)
+            left = np.array(left_sequence[left_start:left_stop])
+
+            right_start = max(0, i - left_length)
+            right_stop = min(right_length, i)
+            right = np.array(right_sequence[right_start:right_stop])
+
+            # We can only match subsequences of the same size.
+            assert len(left) == len(right)
```

Comment:
No new asserts pretty please!

Reply:
Yes. I desperately need it while iterating as any breaking of this is hard to see, and breaks the algorithm very heavily. I'm still undecided wether to include that or not as a strong error. (it **should** not occur, but then the algorithm can still technically function without this but can be dramatically subpar...

Reply:
You cna include an error, just with a test and a raise XxxError.